extensions/sql/data/JSONSQLWriter.cpp

```diff
@@ -0,0 +1,156 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "JSONSQLWriter.h"
+#include "rapidjson/writer.h"
+#include "rapidjson/stringbuffer.h"
+#include "rapidjson/prettywriter.h"
+#include "Exception.h"
+#include "Utils.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace sql {
+
+JSONSQLWriter::JSONSQLWriter(const soci::rowset<soci::row> &rowset, std::ostream *out, MaxCollector* pMaxCollector)
+  : SQLWriter(rowset), json_payload_(rapidjson::kArrayType), output_stream_(out), pMaxCollector_(pMaxCollector) {
+}
+
+JSONSQLWriter::~JSONSQLWriter() {
+}
+
+bool JSONSQLWriter::addRow(const soci::row &row, size_t rowCount) {
+  rapidjson::Document::AllocatorType &alloc = json_payload_.GetAllocator();
+  rapidjson::Value rowobj(rapidjson::kObjectType);
+
+  // 'countColumnsInMaxCollector' is used to check that all columns in maxCollector are in row columns.
+  // It is checked here since don't know if it is possible in 'soci' to get coulmns info of a select statements without executing query.
+  int countColumnsInMaxCollector = 0;
```

Comment:
This - and all related variables - should be size_t, as they cannot be negative and we derive it from a size_t value (row.size()).

Reply:
Fixed.